result-formatter/main.go

```diff
@@ -68,27 +79,53 @@ func main() {
 			break
 		}
 		for _, doc := range result.ResultDocuments {
-			filename, _ := doc.Metadata["workspaceFileName"].(string)
-			if _, ok := fullyFetched[filename]; ok {
+			filename := doc.Metadata.WorkspaceFileName
+
+			// We parse the location regardless of the file potentially being fully fetched already to preserve the
+			// source reference metadata (i.e. where in the document the information was found).
+			// This is a UX thing to help users with manual proofreading of answers.
+			var location string
+			if doc.Metadata.Pages != "" {
+				location = "Pages " + doc.Metadata.Pages
+			} else if doc.Metadata.Page > 0 {
+				location = fmt.Sprintf("Page %d", doc.Metadata.Page)
+			}
+			if location != "" && doc.Metadata.TotalPages > 0 {
+				location = fmt.Sprintf("%s of %d", location, doc.Metadata.TotalPages)
+				_, _ = fmt.Fprintf(os.Stderr, "result doc in file %q at %q\n", filename, location)
```

Comment:
was this done only for debugging or should it be left in?

Reply:
It's debug logs here and below. I just left them in as they shouldn't do any harm going to stderr.
I can remove them if needed though (I hope I won't need them anymore :grimacing: )